ssh/tailssh/incubator.go

```diff
@@ -106,48 +107,41 @@ func (ss *sshSession) newIncubatorCommand() (cmd *exec.Cmd) {
 	incubatorArgs := []string{
 		"be-child",
 		"ssh",
+		"--login-shell=" + lu.LoginShell(),
 		"--uid=" + lu.Uid,
 		"--gid=" + lu.Gid,
-		"--groups=" + gids,
+		"--groups=" + groups,
 		"--local-user=" + lu.Username,
 		"--remote-user=" + remoteUser,
 		"--remote-ip=" + ci.src.Addr().String(),
 		"--has-tty=false", // updated in-place by startWithPTY
 		"--tty-name=",     // updated in-place by startWithPTY
 	}
 
+	forceV1Behavior := ss.conn.srv.lb.NetMap().SelfNode.HasCap(tailcfg.NodeAttrSSHBehaviorV1)
```

Comment:
might want a method on LocalBackend instead so you can easily guard against the small window where NetMap can turn nil on a logout/switch concurrent with an incoming SSH connection and then crash here on the SelfNode deref.

Reply:
Ah, nice! Done. The `HasCap` already guards agains a null SelfNode, but `NetMap()` can be nil, so your suggestion fixes that.